tools/resourcedocsgen/cmd/docs/registry.go

```diff
@@ -39,6 +40,76 @@ import (
 	concpool "github.com/sourcegraph/conc/pool"
 )
 
+// PackageMetadataProvider is an interface for providers that retrieve package metadata
+// from either the filesystem directory or the Pulumi Registry API.
+type PackageMetadataProvider interface {
```

Comment:
Let's put `PackageMetadataProvider` and it's implementors into their own file.

I wouldn't be sad if it was their own package.